packages/tests/tests/filter/eventListener.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { DefaultPubsubTopic } from "@waku/core";
+import type { LightNode } from "@waku/interfaces";
+import { expect } from "chai";
+
+import { MessageCollector, NimGoNode, tearDownNodes } from "../../src/index.js";
+
+import { runNodes, TestDecoder } from "./utils.js";
+
+describe("Waku Filter V2: Susbcription: Event Listener", function () {
+  this.timeout(10000);
+  let waku: LightNode;
+  let nwaku: NimGoNode;
+  let collector: MessageCollector;
+
+  this.beforeEach(async function () {
+    this.timeout(15000);
+    [nwaku, waku] = await runNodes(this, [DefaultPubsubTopic]);
+    collector = new MessageCollector();
+  });
+
+  this.afterEach(async function () {
+    this.timeout(15000);
+    await tearDownNodes(nwaku, waku);
+  });
+
+  describe("should fail for invalid key", function () {
+    it("key: number", async function () {
+      const subscription = await waku.filter.createSubscription([TestDecoder]);
+      expect(() => {
+        subscription.addEventListener(1, collector.filterCallback);
+      }).to.throw();
+    });
+    it("key: random string", async function () {
+      const subscription = await waku.filter.createSubscription([TestDecoder]);
+      expect(() => {
+        subscription.addEventListener("fail", collector.filterCallback);
+      }).to.throw();
+    });
+    it("key: invalid content topic", async function () {
+      const subscription = await waku.filter.createSubscription([TestDecoder]);
+      expect(() => {
+        subscription.addEventListener(
+          "/waku/2/default-waku/test",
+          collector.filterCallback
+        );
+      }).to.throw();
+    });
+  });
+  describe("should pass for a valid key", function () {
+    it("key: valid pubsub topic", async function () {
+      const subscription = await waku.filter.createSubscription([TestDecoder]);
+      expect(() => {
+        subscription.addEventListener(
+          TestDecoder.contentTopic,
+          collector.filterCallback
+        );
+      }).to.not.throw();
+    });
+  });
```

Comment:
These tests can leave in filter.spec.ts near the filter.ts file as it does not required a nwaku node.

Reply:
this is tricky

if we follow how it works with DOM events - we should just ignore anything - no throw, no error
another way of handling error could be invoking callback with error object - which does not seem feasible in that case too

so the only ok solution seems to throw if it is not possible to subscribe to such event (e.g no content topic)

Reply:
we are still creating a waku node for which `sdk` will need to be added as a dependency in `core`